### PR TITLE
Refactor readiness probe check

### PR DESCRIPTION
**Pull request by artem-belov**
## Description
Readiness probe should be success only when all services are started.
Restart container if starting services returns error.

## Motivation and Context
Readiness probe was success even if services was not ready yet.

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [x] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: controlplane/cmd/nsmd/main.go [hunk 1]
```diff
@@ -54,25 +57,34 @@ func main() {
 	logrus.Info("Starting Dataplane registration server...")
 	if err := server.StartDataplaneRegistratorServer(); err != nil {
```
Comment (lobkovilya) [inline controlplane/cmd/nsmd/main.go]: Why `quit` is not passed to `StartDataplaneRegistratorServer`?

File: test/integration/basic_nsmd_deploy_test.go [hunk 2]
```diff
@@ -39,16 +41,35 @@ func testNSMgrDdataplaneDeploy(t *testing.T, nsmdPodFactory func(string, *v1.Nod
...
 		logrus.Printf("At least two Kubernetes nodes are required for this test")
-		Expect(len(nodes)).To(Equal(2))
+		Expect(len(nodes)).To(Equal(nodesCount))
 		return
 	}
 
-	_, err = kubetest.SetupNodes(k8s, 2, defaultTimeout)
-	Expect(err).To(BeNil())
+	var wg sync.WaitGroup
+	for i := 0; i < nodesCount; i++ {
```
Comment (lobkovilya) [inline test/integration/basic_nsmd_deploy_test.go]: Seems like nodes are independent in that test. Is there any point to do the same checks for more than one node?

File: test/integration/basic_nsmd_deploy_test.go [hunk 3]
```diff
@@ -15,15 +17,15 @@ import (
...
 }
 
-func TestNSMgrDdataplaneDeployLiveCheck(t *testing.T) {
-	testNSMgrDdataplaneDeploy(t, pods.NSMgrPodLiveCheck, pods.ForwardingPlaneWithLiveCheck)
+func TestNSMgrDataplaneDeployLiveCheck(t *testing.T) {
+	testNSMgrDataplaneDeploy(t, 2, pods.NSMgrPodLiveCheck, pods.ForwardingPlaneWithLiveCheck)
 }
 
-func testNSMgrDdataplaneDeploy(t *testing.T, nsmdPodFactory func(string, *v1.Node, string) *v1.Pod, dataplanePodFactory func(string, *v1.Node, string) *v1.Pod) {
+func testNSMgrDataplaneDeploy(t *testing.T, nodesCount int, nsmdPodFactory func(string, *v1.Node, string) *v1.Pod, dataplanePodFactory func(string, *v1.Node, string) *v1.Pod) {
```
[the remark this answers is not in the document]
Comment (artem-belov) [reply test/integration/basic_nsmd_deploy_test.go]: I think it looks better when it is configurable number even if we do not use another values.

File: controlplane/pkg/nsmd/nsmd.go [hunk 4]
```diff
@@ -387,7 +387,9 @@ func (nsm *nsmServer) Stop() {
 	}
 }
 
-func StartNSMServer(model model.Model, manager nsm.NetworkServiceManager, serviceRegistry serviceregistry.ServiceRegistry, apiRegistry serviceregistry.ApiRegistry) (NSMServer, error) {
+// StartNSMServer registers and starts gRPC server which is listening for
+//// Network Service requests.
```
[the remark this answers is not in the document]
Comment (artem-belov) [reply controlplane/pkg/nsmd/nsmd.go]: Fixed

**Pull request merged**